tests/MongoDB.Driver.Core.Tests/Core/Clusters/DnsClientWrapperTests.cs

```diff
@@ -26,38 +27,28 @@ namespace MongoDB.Driver.Core.Clusters
     public class DnsClientWrapperTests
     {
         [Fact]
-        public void constructor_should_initialize_instance()
+        public void Instance_should_return_the_same_instance_each_time()
         {
-            var subject = new DnsClientWrapper();
-
-            subject.LookupClient.Should().NotBeNull();
-        }
-
-        [Fact]
-        public void LookupClient_should_return_expected_result()
-        {
-            var subject = new DnsClientWrapper();
-
-            var result = subject.LookupClient;
-
-            result.Should().NotBeNull();
+            var instance1 = DnsClientWrapper.Instance;
+            var instance2 = DnsClientWrapper.Instance;
+            instance1.Should().BeSameAs(instance2);
         }
 
         [Theory]
         [InlineData("_mongodb._tcp.test5.test.build.10gen.cc", new[] { "localhost.test.build.10gen.cc.:27017" }, false)]
         [InlineData("_mongodb._tcp.test5.test.build.10gen.cc", new[] { "localhost.test.build.10gen.cc.:27017" }, true)]
         public void ResolveSrvRecords_should_return_expected_result(string service, string[] expectedEndPoints, bool async)
         {
-            var subject = new DnsClientWrapper();
+            var subject = CreateSubject();
 
             List<SrvRecord> result;
             if (async)
             {
-                result = subject.ResolveSrvRecords(service, CancellationToken.None);
+                result = subject.ResolveSrvRecordsAsync(service, CancellationToken.None).GetAwaiter().GetResult();
```

Comment:
Good catch!

Of course the tests were passing but in the event of a failure it would have been misleading.